### PR TITLE
Add KEP 2464: Kubetest2 CI migration

### PR DESCRIPTION
**Pull request by amwat**
ref: #2464 

/cc @spiffxp @BenTheElder @saschagrunert

### CODE REVIEW

File: keps/sig-testing/2464-kubetest2-ci-migration/README.md
Comment (saschagrunert) [inline keps/sig-testing/2464-kubetest2-ci-migration/README.md:231]:
```suggestion
- Create shadow canary jobs for each [presubmit-kubernetes-blocking](https://testgrid.k8s.io/presubmits-kubernetes-blocking), [release-blocking](https://testgrid.k8s.io/sig-release-master-blocking) to use kubetest2 as optional, non-blocking
```

File: keps/sig-testing/2464-kubetest2-ci-migration/README.md
Comment (saschagrunert) [inline keps/sig-testing/2464-kubetest2-ci-migration/README.md:274]: @hasheddan we probably should target a v1.0.0/v1 release of k/release to freeze our API.

**Review by saschagrunert — COMMENTED** [review]
This is great to see, thank you @amwat! :pray:

File: keps/sig-testing/2464-kubetest2-ci-migration/README.md
Comment (hasheddan) [reply keps/sig-testing/2464-kubetest2-ci-migration/README.md:274]: @saschagrunert would that be required as part of this KEP?

Comment (saschagrunert) [reply keps/sig-testing/2464-kubetest2-ci-migration/README.md:274]: I think we can keep this a separate topic in k/release, WDYT?

File: keps/sig-testing/2464-kubetest2-ci-migration/README.md
Comment (spiffxp) [inline keps/sig-testing/2464-kubetest2-ci-migration/README.md:132]: I suggested answers for the PRR
```suggestion
- [x] (R) Production readiness review completed
```

File: keps/sig-testing/2464-kubetest2-ci-migration/README.md
Comment (spiffxp) [inline keps/sig-testing/2464-kubetest2-ci-migration/README.md:538]:
```suggestion
n/a (this is a build/test change for the kubernetes project, maintains parity with existing merge-blocking and release-blocking jobs)
```

File: keps/sig-testing/2464-kubetest2-ci-migration/README.md
Comment (spiffxp) [inline keps/sig-testing/2464-kubetest2-ci-migration/README.md:557]:
```suggestion
n/a (this is a build/test change for the kubernetes project, maintains parity with existing merge-blocking and release-blocking jobs)
```

File: keps/sig-testing/2464-kubetest2-ci-migration/README.md
Comment (spiffxp) [inline keps/sig-testing/2464-kubetest2-ci-migration/README.md:589]:
```suggestion
n/a (this is a build/test change for the kubernetes project, not relevant to monitoring production clusters)
```

File: keps/sig-testing/2464-kubetest2-ci-migration/README.md
Comment (spiffxp) [inline keps/sig-testing/2464-kubetest2-ci-migration/README.md:608]:
```suggestion
n/a (this is a build/test change for the kubernetes project, maintains parity with existing merge-blocking and release-blocking jobs)
```

File: keps/sig-testing/2464-kubetest2-ci-migration/README.md
Comment (spiffxp) [inline keps/sig-testing/2464-kubetest2-ci-migration/README.md:659]:
```suggestion
n/a (this is a build/test change for the kubernetes project, maintains parity with existing merge-blocking and release-blocking jobs, including scalability jobs)
```

File: keps/sig-testing/2464-kubetest2-ci-migration/README.md
Comment (spiffxp) [inline keps/sig-testing/2464-kubetest2-ci-migration/README.md:686]:
```suggestion
n/a (this is a build/test change for the kubernetes project, not relevant to troubleshooting problems in production cluster)
```

File: keps/sig-testing/2464-kubetest2-ci-migration/README.md
Comment (spiffxp) [inline keps/sig-testing/2464-kubetest2-ci-migration/README.md:385]: Taking a guess at where you'd like the kubetest2 dependency strategies and doc
```suggestion
- This will be declared **alpha** when:
    - Jobs fetch latest pre-built kubetest2 binaries at runtime
```

File: keps/sig-testing/2464-kubetest2-ci-migration/README.md
Comment (spiffxp) [inline keps/sig-testing/2464-kubetest2-ci-migration/README.md:394]:
```suggestion
- This will be declared **GA** when:
    - Jobs use pre-installed tagged versions of kubetest2 binaries
    - Tracking issues exist for each SIG to migrate their jobs from kubetest to kubetest2 
```

File: keps/sig-testing/2464-kubetest2-ci-migration/README.md
Comment (spiffxp) [inline keps/sig-testing/2464-kubetest2-ci-migration/README.md:390]: I sorta wanted to suggest this in alpha but leave it to you
```suggestion
- This will be declared **beta** when:
    - A guide for migration from kubetest to kubetest2 exists
```

File: keps/sig-testing/2464-kubetest2-ci-migration/README.md
Comment (spiffxp) [inline keps/sig-testing/2464-kubetest2-ci-migration/README.md:210]: /cc @justaugustus
not blocking but wanted to confirm this was accurate

File: keps/sig-testing/2464-kubetest2-ci-migration/README.md
Comment (spiffxp) [inline keps/sig-testing/2464-kubetest2-ci-migration/README.md:170]:
```suggestion
Kubetest and scenarios have both been [deprecated](https://github.com/kubernetes/test-infra/tree/master/kubetest#deprecation-notice), [(2)](https://github.com/kubernetes/test-infra/tree/master/scenarios#deprecation-notice) for a while, with a recommendation to move jobs to [kubetest2](https://github.com/kubernetes-sigs/kubetest2). 
```

File: keps/sig-testing/2464-kubetest2-ci-migration/README.md
Comment (spiffxp) [inline keps/sig-testing/2464-kubetest2-ci-migration/README.md:189]:
```suggestion
- Document guides for migrating jobs away from kubetest to kubetest2 to enable self-service migration of non-blocking jobs
```

File: keps/sig-testing/2464-kubetest2-ci-migration/kep.yaml
Comment (spiffxp) [inline keps/sig-testing/2464-kubetest2-ci-migration/kep.yaml:16]:
```suggestion
- johnbelamaric 
```

File: keps/sig-testing/2464-kubetest2-ci-migration/kep.yaml
Comment (spiffxp) [inline keps/sig-testing/2464-kubetest2-ci-migration/kep.yaml:9]:
```suggestion
creation-date: 2021-02-08
```

File: keps/sig-testing/2464-kubetest2-ci-migration/kep.yaml
Comment (spiffxp) [inline keps/sig-testing/2464-kubetest2-ci-migration/kep.yaml:7]: /cc @wojtek-t 
we'd like your input to move this to implementable
```suggestion
- sig-release
- sig-scalability
```

**Review by spiffxp — CHANGES_REQUESTED** [review]
/approve
As SIG Testing approver.

/hold
But I have some suggested changes I'd like to see before /lgtm

File: .gitignore
Comment (spiffxp) [inline .gitignore:27]: Also, can you drop this change? This is requiring root approval

Comment (amwat) [reply .gitignore]: done

**Review by spiffxp — APPROVED** [review]
/lgtm

**Pull request merged**
